modules/openapi-generator/src/main/resources/python/configuration.mustache

```diff
@@ -212,7 +214,8 @@ conf = {{{packageName}}}.Configuration(
         self.logger_file = None
         """Debug file location
         """
-        self.debug = False
+        if debug is not None:
+            self.debug = debug
```

Comment:
If this `self.debug` property is not always set, then [the underlying `__debug` property ](https://github.com/OpenAPITools/openapi-generator/blob/c12633fe80ae6d2f51ca27626c3ddf3be823e8b4/samples/client/echo_api/python-disallowAdditionalPropertiesIfNotPresent/openapi_client/configuration.py#L291-L320) will also not always be set, which may produce `AttributeError`.

I would suggest to at least set `self.__debug` to `False` by default so that the `debug` setter/getter don't have a way to fail, and all the code that are also using this attribute will also not fail.

Reply:
I can easily add a default value for `self.__debug`, but I'm not sure how to write the tests.
Can you point me to a sample I can adapt?

Reply:
You can check here for instance: https://github.com/OpenAPITools/openapi-generator/blob/master/samples/openapi3/client/petstore/python/tests/test_configuration.py

There are several tests around this class, you should be able to create a test that fails with your current change and make it pass by setting an initial value for `__debug`.

There may be other places with this kind of tests (they are written manually), but I didn't check.

Reply:
Uhm... I was starting to write some tests, but it seems there are some unrelated failures.

```bash
$ cd samples/openapi3/client/petstore/python
$ poetry install
[...]
$ poetry run pytest
[...]
================================================= short test summary info ==================================================
FAILED tests/test_model.py::ModelTests::test_valdiator - AttributeError: 'ModelTests' object has no attribute 'assertEquals'. Did you mean: 'assertEqual'?
FAILED tests/test_pet_api.py::PetApiTests::test_add_pet_and_get_pet_by_id - urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='localhost', port=80): Max retries exceeded with url: /v2/pet...
FAILED tests/test_pet_api.py::PetApiTests::test_add_pet_and_get_pet_by_id_with_http_info - urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='localhost', port=80): Max retries exceeded with url: /v2/pet...
FAILED tests/test_pet_api.py::PetApiTests::test_add_pet_and_get_pet_by_id_without_preload_content_flag - urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='localhost', port=80): Max retries exceeded with url: /v2/pet...
FAILED tests/test_pet_api.py::PetApiTests::test_delete_pet - urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='localhost', port=80): Max retries exceeded with url: /v2/pet...
FAILED tests/test_pet_api.py::PetApiTests::test_find_pets_by_status - urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='localhost', port=80): Max retries exceeded with url: /v2/pet...
FAILED tests/test_pet_api.py::PetApiTests::test_find_pets_by_tags - urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='localhost', port=80): Max retries exceeded with url: /v2/pet...
FAILED tests/test_pet_api.py::PetApiTests::test_get_pet_by_id_serialize - urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='localhost', port=80): Max retries exceeded with url: /v2/pet...
FAILED tests/test_pet_api.py::PetApiTests::test_update_pet - urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='localhost', port=80): Max retries exceeded with url: /v2/pet...
FAILED tests/test_pet_api.py::PetApiTests::test_update_pet_with_form - urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='localhost', port=80): Max retries exceeded with url: /v2/pet...
FAILED tests/test_pet_api.py::PetApiTests::test_upload_file - urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='localhost', port=80): Max retries exceeded with url: /v2/pet...
================================== 11 failed, 264 passed, 3 skipped, 9 warnings in 14.25s ==================================
```

I'm normally using python 3.12, where [it was removed](https://docs.python.org/3.12/whatsnew/3.12.html#id3)
Should I also update the unrelated tests?

Reply:
* For `assertEquals`, this should be migrated in another PR I think. Can you test with 3.11 instead?
* For the URL failures, there's an explanation how to start a test container to support the tests at the top of `tests/test_pet_api.py`

Reply:
(oh, and for the various `PetApiTests` failed tests, while it's easy to follow the instructions in the tests https://github.com/OpenAPITools/openapi-generator/blob/793aba724abe2c9783811f0c580d71d3fec06fe1/samples/openapi3/client/petstore/python/tests/test_pet_api.py#L6-L11 I would suggest to enhance them with [testcontainers](https://testcontainers.com/) so you can avoid launching the docker image by hand